src/main/row.ts

```diff
@@ -210,44 +211,34 @@ function toHexString(byteArray: number[]): string {
  * objects are constructed for both the leads and the helpers
  * 
  * @param row Row to construct and retrieve BasecampTodoRequest objects for
- * @returns array of BasecampTodoRequest objects
+ * @returns a map associating role titles with BasecampTodoRequest objects
  */
-export function getBasecampTodoRequestsForRow(row: Row): BasecampTodoRequest[] {
-    let basecampTodoRequests: BasecampTodoRequest[] = [];
-    const leadsBasecampTodoRequest: BasecampTodoRequest | undefined = getBasecampTodoForLeads(row);
-
-    if(leadsBasecampTodoRequest !== undefined) {
-        basecampTodoRequests.push(leadsBasecampTodoRequest);
-    }
-
-    const helpersBasecampTodoRequest: BasecampTodoRequest[] = getBasecampTodosForHelpers(row);
-
-    if(helpersBasecampTodoRequest.length > 0) {
-        basecampTodoRequests = basecampTodoRequests.concat(helpersBasecampTodoRequest);
-    }
-
+export function getBasecampTodoRequestsForRow(row: Row): Map<string, BasecampTodoRequest> {
+    let basecampTodoRequests = new Map<string, BasecampTodoRequest>();
+    getBasecampTodoForLeads(row, basecampTodoRequests);
+    getBasecampTodosForHelpers(row, basecampTodoRequests);
     return basecampTodoRequests;
 }
 
 /**
  * Constructs a BasecampTodoRequest object for the lead of a given row
  * 
  * @param row row to construct the BasecampTodoRequest for
- * @returns BasecampTodoRequest
+ * @param basecampTodoRequests the Map that associates the roleTitle with the request
  */
-export function getBasecampTodoForLeads(row: Row): BasecampTodoRequest | undefined {
-    const basecampTodoContent: string = `Lead: ${row.what.value}`;
+export function getBasecampTodoForLeads(row: Row, basecampTodoRequests: Map<string, BasecampTodoRequest>): void {
+
+    const roleTitle = "Lead"
```

Comment:
let's make this a constant defined at the top of this class

Reply:
It's done